### ISSUE TITLE
On Windows isdir throws when arg is an URL breaking adding of non-registered packages

### ISSUE THREAD

**Issue by krcools** [JuliaLang/Pkg.jl#322]
Adding a package by specifying the URL of a non-registered repo breaks on windows. The problem can be traced back to `isdir` throwing (instead of returning false) when being fed a URL. This breaks the code at:

https://github.com/JuliaLang/Pkg.jl/blob/4ef95bfaa15125b254954db4f28b2a033dc11d25/src/REPLMode.jl#L141

Terminal output:

```
julia> Pkg.REPLMode.isdir("https://github.com/krcools/BEAST.jl")
ERROR: stat: invalid argument (EINVAL)
Stacktrace:
 [1] stat(::String) at .\stat.jl:68
 [2] isdir(::String) at .\stat.jl:303
 [3] top-level scope
```

**Comment by KristofferC** [comment]
Thanks for the report, should be fixed by https://github.com/JuliaLang/Pkg.jl/pull/323.

**Comment by KristofferC** [comment]
Closed bt #323
